frontend/ui/src/app/deployments/deployment-form/deployment-form.component.ts

```diff
@@ -247,82 +227,63 @@ export class DeploymentFormComponent implements OnInit, AfterViewInit, OnDestroy
       if (id) {
         this.deployForm.controls.applicationId.disable();
         this.deployForm.controls.swarmMode.disable();
         this.deployForm.controls.forceRestart.enable();
       } else {
-        this.deployForm.controls.applicationId.enable();
+        if (!this.disableApplicationSelect()) {
+          this.deployForm.controls.applicationId.enable();
+        }
         this.deployForm.controls.swarmMode.enable();
         this.deployForm.controls.forceRestart.disable();
       }
     });
 
-    combineLatest([
-      this.applicationId$,
-      this.applicationVersionId$,
-      this.deploymentTarget$.pipe(distinctUntilChanged((a, b) => a?.id === b?.id)),
-      this.deployment$.pipe(
-        distinctUntilChanged((a, b) => a?.envFileData === b?.envFileData && a?.valuesYaml === b?.valuesYaml)
-      ),
-    ])
+    combineLatest([this.applicationId$, this.applicationVersionId$, this.deploymentId$])
       .pipe(
         debounceTime(5),
-        switchMap(([applicationId, versionId, dt, d]) =>
-          combineLatest([
-            of(dt),
-            // Only fill in the template if there is no existing deployment or the existing deployment has no values/env file
-            versionId && applicationId && !(d?.valuesYaml || d?.envFileData)
-              ? this.applications.getTemplateFile(applicationId, versionId).pipe(catchError(() => NEVER))
-              : NEVER,
-          ])
+        switchMap(([applicationId, versionId, deploymentId]) =>
+          versionId && applicationId && !deploymentId
+            ? this.applications.getTemplateFile(applicationId, versionId).pipe(catchError(() => NEVER))
+            : NEVER
         ),
         takeUntil(this.destroyed$)
       )
-      .subscribe(([deploymentTarget, templateFile]) => {
-        if (deploymentTarget) {
-          if (deploymentTarget.type === 'kubernetes') {
-            this.deployForm.controls.valuesYaml.patchValue(templateFile ?? '');
-          } else {
-            this.deployForm.controls.envFileData.patchValue(templateFile ?? '');
-          }
+      .subscribe((templateFile) => {
+        const type = this.deploymentType();
+        if (type === 'kubernetes') {
+          this.deployForm.controls.valuesYaml.patchValue(templateFile ?? '');
+        } else {
+          this.deployForm.controls.envFileData.patchValue(templateFile ?? '');
         }
       });
 
-    combineLatest([
-      this.applicationId$,
-      this.applicationVersionId$,
-      this.deploymentTarget$.pipe(distinctUntilChanged((a, b) => a?.id === b?.id)),
-    ])
+    combineLatest([this.applicationId$, this.applicationVersionId$])
       .pipe(
         debounceTime(5),
-        switchMap(([applicationId, versionId, dt]) =>
-          versionId && applicationId && dt?.type === 'docker'
+        switchMap(([applicationId, versionId]) =>
+          versionId && applicationId && this.deploymentType() === 'docker'
```

Comment:
I think `this.deploymentType` should be added to the `combineLatest` using the `asObservable` utility, so that changes are registered correctly